source/Fleet.cpp

```diff
@@ -493,10 +494,13 @@ void Fleet::Place(const System &system, list<shared_ptr<Ship>> &ships, bool carr
 // Do the randomization to make a ship enter or be in the given system.
 const System *Fleet::Enter(const System &system, Ship &ship, const System *source)
 {
-	bool canEnter = (source != nullptr);
-	if(!canEnter && any_of(system.Links().begin(), system.Links().end(),
-			[ship](const System *link) noexcept -> bool { return ship.GetGovernment()->AllowJumpingTo(*link); }))
-		canEnter = true;
+	bool unconstrained = ship.GetPersonality().IsUnconstrained();
+	bool canEnter = (source != nullptr && any_of(system.Links().begin(), system.Links().end(),
```

Comment:
```suggestion
	bool canEnter = (source != nullptr || any_of(system.Links().begin(), system.Links().end(),
```

You changed the logic accidentally.